sale_order_line_packaging_qty/models/sale_order_line.py

```diff
@@ -66,10 +66,16 @@ def _onchange_product_packaging_qty(self):
     @api.onchange("product_packaging")
     def _onchange_product_packaging(self):
         if self.product_packaging:
+            pack_qty = 1
```

Comment:
@TDu I suppose that the play_onchanges is called on a record initialized with self.new()... If it's the case, to be consistent with odoo, the self.new() should be called with the origin args.... see https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5510.

Reply:
@lmignon Thanks for your answer, from what I can tell the sale.order is created with the `create` function. Done in an other OCA module https://github.com/OCA/edi/blob/cc04afc899246919a4ac90373d92ced0a398641c/sale_order_import/wizard/sale_order_import.py#L248

Reply:
Something is missing into this puzzle... Nevertheless your proposed change makes sense even if I would prefer to not rely on some implicit rules to determine the context in which the onchange method is called. I'm curious to discover the method triggering the call to this onchange....

Reply:
`play_onchanges` comes from: https://github.com/OCA/server-tools/blob/5e432fdd9b33a18badff88119e9d8f0791736369/onchange_helper/models/base.py#L24

Is it possible that we have no `_origin` when going through this method?

Reply:
@guewen Indeed, IMO, whe should initialize the "_origin" field on the current recordset before the call to `onchange` In this way we are consistent with the way the onchange methods are called from the UI. https://github.com/OCA/server-tools/blob/13.0/onchange_helper/models/base.py#L57
somethng like:

```python
    ....
     with self._set_origin():
         for field in onhcange_fields:
            onchange_values = self.onchange(all_values, field, onchange_specs)
    ...

@contextmanager
def _set_origin(self):
    if self:
        self.ensure_one()
        setattr(self, '_origin', self)
    yield
    if self:
        delattr(self, "_origin")
```

Reply:
@guewen I'm not sure this code will work in odoo 13. The poposed PR is probably the safer way to solve your issue

Reply:
Reading the code in models.py for odoo13. I'm sure that the proposed change will not work.

Reply:
@TDu I tested with sale_order_import and the qty is wrong and/or the deducted packaging is wrong.

In my tests, I have a product with a retail box=22 units and transport box of 176 units.
Test 1: I order a qty not multiple of a packaging. Then SO line qty is always 1 retail box instead of the order qty
Test 2: I order a qty multiple of a transport box. Then SO line packaging is not expressed with the biggest packaging (transport box) but as a multiple of retail box.

When doing those 2 tests with odoo SO interface, it works properly

Reply:
@jbaudoux In the UI if the quantity is set first (with not a multiple of packaging quantity) and then the product is set, then the quantity is lost and 1 retail box is selected and the quantity contains the retail box quantity.

Reply:
This is what happens on the import. To change this behaviour would mean that a default package is not selected automatically. As by default Odoo sets 1 as a quantity I see no way to know if this is a default 1 or required 1 on the line.

Reply:
@TDu I would remove the default packaging. It is not convenient in the UI as you do not see the possibilities without emptying the field. So better leave it empty by default and the user is free to select one. It will also be easier this way if the user does not want to select one as emptying that field is not that easy in the UI (it sometimes comes back).

Reply:
@jbaudoux I checked the code and the change happens in this onchange https://github.com/OCA/sale-workflow/blob/35feda502a35c55f0074232da1041926c0277902/sale_by_packaging/models/sale_order_line.py#L56
So only if the product is to be sold only by packaging. (This was the case for the product I used for my test).
Disabling the `Only sell by packaging`  on the product and the problem is gone.
So it is actually a feature of the module.

Reply:
sell_only_by_packaging should require a packaging in the UI but not fill with default value

Reply:
What is the status here? Is there something to change or is this thread resolved?

Reply:
Yes by this https://github.com/OCA/sale-workflow/pull/1276/commits/6c291697db159c6c7d518526bae1eba43b9c7ccc

Reply:
It is a commit below

Reply:
The requirement is done by a constraint